providers/apis/defi/raydium/provider.go

```diff
@@ -0,0 +1,253 @@
+)
+
+var _ handlers.APIFetcher[mmtypes.Ticker, *big.Int] = (*APIPriceFetcher)(nil)
+
+// SolanaJSONRPCClient is the expected interface for a solana JSON-RPC client according
+// to the APIPriceFetcher.
+//
+//go:generate mockery --name SolanaJSONRPCClient --output ./mocks/ --case underscore
+type SolanaJSONRPCClient interface {
+	GetMultipleAccountsWithOpts(
+		ctx context.Context,
+		accounts []solana.PublicKey,
+		opts *rpc.GetMultipleAccountsOpts,
+	) (out *rpc.GetMultipleAccountsResult, err error)
+}
+
+// APIPriceFetcher is responsible for interacting with the solana API and querying information
+// about the price of a given currency pair.
+type APIPriceFetcher struct {
+	// market represents the ticker configurations for this provider.
+	market oracletypes.ProviderMarketMap
+
+	// config is the APIConfiguration for this provider
+	config config.APIConfig
+
+	// client is the solana JSON-RPC client used to query the API.
+	client SolanaJSONRPCClient
+
+	// metaDataPerTicker is a map of ticker.String() -> TickerMetadata
+	metaDataPerTicker map[string]TickerMetadata
+
+	// logger
+	logger *zap.Logger
+}
+
+// NewAPIPriceFetcher returns a new APIPriceFetcher. This method constructs the
+// default solana JSON-RPC client in accordance with the config's URL param.
+func NewAPIPriceFetcher(
+	market oracletypes.ProviderMarketMap,
+	config config.APIConfig,
+	logger *zap.Logger,
+) (*APIPriceFetcher, error) {
+	return NewAPIPriceFetcherWithClient(
+		market,
+		config,
+		rpc.New(config.URL),
+		logger,
+	)
+}
+
+// NewAPIPriceFetcherWithClient returns a new APIPriceFetcher. This method requires
+// that the given market + config are valid, otherwise a nil implementation + an error
+// will be returned.
+func NewAPIPriceFetcherWithClient(
+	market oracletypes.ProviderMarketMap,
+	config config.APIConfig,
+	client SolanaJSONRPCClient,
+	logger *zap.Logger,
+) (*APIPriceFetcher, error) {
+	if err := config.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("config for raydium is invalid: %w", err)
+	}
+
+	if err := market.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("market config for raydium is invalid: %w", err)
+	}
+
+	// check fields of config
+	if config.Name != Name {
+		return nil, fmt.Errorf("config.Name is not %s", Name)
+	}
+
+	if market.Name != Name {
+		return nil, fmt.Errorf("market.Name is not %s", Name)
+	}
+
+	if !config.Enabled {
+		return nil, fmt.Errorf("config is not enabled")
+	}
+
+	// generate metadata per ticker
+	metadataPerTicker := make(map[string]TickerMetadata)
+	for _, ticker := range market.OffChainMap {
+		metadata, err := unmarshalMetadataJSON(ticker.Metadata_JSON)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshalling metadata for ticker %s: %w", ticker.String(), err)
+		}
+
+		if err := metadata.ValidateBasic(); err != nil {
+			return nil, fmt.Errorf("metadata for ticker %s is invalid: %w", ticker.String(), err)
+		}
+
+		metadataPerTicker[ticker.String()] = metadata
```

Comment:
Is there a case where there is a duplicate `ticker` in OffchainMap?  Then there would be an overwrite into this map